x-pack/plugins/fleet/server/services/artifacts/artifacts.ts

```diff
@@ -202,6 +202,39 @@ export const deleteArtifact = async (esClient: ElasticsearchClient, id: string):
   }
 };
 
+export const bulkDeleteArtifacts = async (
+  esClient: ElasticsearchClient,
+  ids: string[]
+): Promise<Error[]> => {
+  try {
+    const body = ids.flatMap((id) => [
+      {
+        delete: { _index: FLEET_SERVER_ARTIFACTS_INDEX, _id: id },
+      },
+    ]);
+
+    const res = await withPackageSpan(`Bulk delete fleet artifacts`, () =>
+      esClient.bulk({
+        body,
+        refresh: 'wait_for',
```

Comment:
If we don't wait for de deletions to be done, then the `clanup` phase will try to remove artifacts that might be removed later. This is why I used here the `wait_for`, otherwise, the `cleanup` method in manifest manager will hit errors because it queried some orphan artifacts (not yet removed from the search) but already removed from the index.

Reply:
Got it. Thanks.